test/gtest/Makefile.am

```diff
@@ -33,8 +33,8 @@ export ASAN_OPTIONS=protect_shadow_gap=0
 
 # LD_PRELOAD puts ASAN first in initial library list which is required for
 # correct ASAN work.
-ASAN_LD_PRELOAD=$(shell ldconfig -p | grep libasan.so | tr ' ' '\n' | grep / | head -1)
-TEST_ENV=LD_PRELOAD=$(ASAN_LD_PRELOAD):$(LD_PRELOAD)
+ASAN_SO=$(shell ldd $(abs_builddir)/gtest | grep libasan.so | tr ' ' '\n' | grep /)
+TEST_ENV=LD_PRELOAD=$(ASAN_SO):$(LD_PRELOAD)
```

Comment:
```suggestion
ASAN_LIB=$(shell ldd $(abs_builddir)/gtest | awk '/libasan.so/ {print $3}')
TEST_ENV=LD_PRELOAD=$(ASAN_LIB):$(LD_PRELOAD)
```

Reply:
Proposed version generates the following LD_PRELOAD string which causes error:
```
env LD_PRELOAD= libasan.so.6 => /lib/aarch64-linux-gnu/libasan.so.6 (0x0000ffff8bb90000):
```

Reply:
maybe need to add some escaping to $3

Reply:
For some reason awk don't work, I applied another similar solution based on `cut` tool usage. Pls check the latest commit to that PR.